src/main/g8/default.properties

```diff
@@ -8,6 +8,7 @@ scala_version = 2.13.8
 other_scala_version = 3.1.1
 sbt_version = 1.6.2
 sbt_typelevel_version = 0.4.7
+jdk_version = 8
```

Comment:
I think this setting could be more smartly integrated into the build.sbt, as well.

JDK8 active support just ended in March 2022, so it also wouldn't be unreasonable to bump this to 11.

Reply:
Yes, it's not unreasonable. See also https://github.com/typelevel/sbt-typelevel/issues/215.

Reply:
I think the answer is `tlJdkRelease`, but I'm not entirely sure what that's going to do to the generated JDKs for CI.  I would think that's for sbt-typelevel concern.  I'd much rather sort that out in Scala than in g8.

Reply:
In the same way that we opt'd to default to Scala 2.13 and 3.1.1 I think we should default to JDK 11

Would JDK 17 just be *too* controversial?

Reply:
With the `tlJdkRelease` setting any JDK should be fine. 17 has it advantages. fs2 needs it for example. And http4s uses some of those features, so it kind of needs JDK 17 too to "fully exercise" its tests 😆

Reply:
I prefer to deploy applications to the latest LTS, which is 17.

I generally prefer to support the earliest thing that works in libraries, which is generally 8.  Getting past 8 is an unusually hard Java upgrade.  But with active support ending a few days ago, I'm done exerting myself to get below 11.

I don't know whether this template is more for apps or libraries, and it may be that we want the devshell JDK to be different from the release JDK and/or the CI set of JDKs.

Reply:
@valencik and I discussed a bit on Discord. I think it makes sense to keep the devshell default in line with the sbt-typelevel default for CI.

In sbt-typelevel 0.5.0 we'll set the `tlJdkRelease` flag to 8 by default. At that point, we can update both the devshell and CI to JDK 17 as suggested in https://github.com/typelevel/sbt-typelevel/issues/215.